### DISCUSSION TITLE
Feature request: support for custom caption rendering

### DISCUSSION THREAD

**Discussion by sampotts** [vimeo/player.js#1050] [Feature requests]
### Expected Behavior

It would be awesome if there was a way that the player.js SDK could support custom caption rendering in that it's possible to call `enableTextTrack` while hiding the default/Vimeo rendering of captions and still getting `cuechange` events. 

### Actual Behavior

I've tried `cc=false` but while I can still get the tracks with getTextTracks, it's not possible to enable a textTrack and get `cuechange` events. 

### Steps to Reproduce

Will create a repro asap.

**Comment by mattdelacdev** [comment]
Is there any update here? @rkrishnan8594

**Comment by rkrishnan8594** [comment]
Thanks for sharing this idea, @sampotts. I agree that this would be a nice feature for the SDK to support. I've converted this to a discussion as it's not a bug report, but I'll ensure this is added to our list of improvements to prioritize.

**Comment by xini** [comment]
@rkrishnan8594, is there an update on this?

**Comment by rkrishnan8594** [comment]
@xini - I'm optimistic that we'll be able to get to this feature request soon, hopefully within the next few weeks.

**Comment by dos-format-d** [comment]
Any updates on this?

**Comment by xini** [comment]
Any updates, @rkrishnan8594?

**Comment by rkrishnan8594** [comment]
I can confirm that we're working on this now and plan to ship support within the next two weeks!

**Comment by xini** [comment]
@rkrishnan8594 that's awesome! Thanks! 🙏

**Comment by jdreetz** [comment]
@xini @dos-format-d @sampotts @mattdelacdev we just released an update to support this - https://github.com/vimeo/player.js?tab=readme-ov-file#enabletexttracklanguage-string-kind-string-showing-boolean-promiseobject-invalidtracklanguageerrorinvalidtrackerrorerror

we exposed a third parameter (`showing`) to the `enableTextTrack` method, that will enable text tracks (cue change events will fire), but we won't render anything in the Vimeo Player UI for them.

**Reply by sampotts** [reply]
> @xini @dos-format-d @sampotts @mattdelacdev we just released an update to support this - https://github.com/vimeo/player.js?tab=readme-ov-file#enabletexttracklanguage-string-kind-string-showing-boolean-promiseobject-invalidtracklanguageerrorinvalidtrackerrorerror
> 
> 
> 
> we exposed a third parameter (`showing`) to the `enableTextTrack` method, that will enable text tracks (cue change events will fire), but we won't render anything in the Vimeo Player UI for them. 

Awesome, thanks!
